hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/ClusterJoinManager.java

```diff
@@ -788,6 +791,10 @@ private void startJoin() {
                     return;
                 }
 
+                if (preJoinOperation != null) {
+                    nodeEngine.getOperationService().run(preJoinOperation);
+                }
```

Comment:
```suggestion
                OnJoinOp preJoinOp = preparePreJoinOps();

                if (preJoinOperation != null) {
                    nodeEngine.getOperationService().run(preJoinOperation);
                }
```

Seems it is possible to send joining member's listener-registrations back to it with `preparePreJoinOps`. The above calling order change suggestion is for this possibility.